### ISSUE TITLE
[Object_State] Use length for z-axis and height for y-axis

### ISSUE THREAD

**Issue by yianni** [strands-project/strands_qsr_lib#105]
Can't stretch enough how **dangerous** this change is.

At the moment [here](https://github.com/strands-project/strands_qsr_lib/blob/master/qsr_lib/src/qsrlib_io/world_trace.py#L42):

```python
self.width = width  # x size
self.length = length  # y size
self.height = height  # z size
```

while it is much more senseful and standard that the y-size is height and the z-size is length.

But, as said at start can't stretch how **dangerous** this change will be to both QSRlib developers and to users to the point that I am tempted to leave as it is for eternity.

Opinions?

**Comment by pet1330** [comment]
Although this is not only a ROS library, it is going to be used prodominently by people using the ROS system, so it might be best to align it with the [ROS Coordinate Frame Conventions](http://wiki.ros.org/geometry/CoordinateFrameConventions#Orientation). Although this may affect some current implementations, it would at least be a standard to follow. 

so:

| Label| ROS Standard | current  | Yianni suggestion |
| :-------------: |:-------------:| :-----:|:-----:
| Width | Y | X | X |
| Length | X  |   Y | Z |
|Height | Z  |    Z | Y |

**Comment by yianni** [comment]
I see the reasoning but I have a feeling this will be enormously confusing especially with vision people.

Of course I am open to suggestions, hence the initial question. Aim is to find the easiest and most "standard" way of representing this (we might already have it).

**Comment by yianni** [comment]
(geometry_msgs)[http://wiki.ros.org/geometry_msgs] could have been a starting point but does not have anything regarding "size".

Also atm (although I don't think anyone has used it so it is safe to change) the fields kept are roll/pitch/yaw, geom_msgs uses quaternions; that might be something to change for combatility.

**Comment by yianni** [comment]
Also see the (PCL API Public Attributes)[http://docs.pointclouds.org/1.7.1/classpcl_1_1_point_cloud.html]

**Comment by yianni** [comment]
Since this is eternally open (but not urgent before v1.0), I will still leave it until we reach a decision.

Another suggestion I have is that instead of using width, length, height we use xsize, ysize, zsize which is pretty clear what is what then.

Possibly also deprecate roll, pitch, yaw as it has the same issue as w,l,h and possibly replace with quartenion values.

**Comment by yianni** [comment]
closed by #177
